bindings.c

```diff
@@ -4046,6 +4046,49 @@ static uint64_t get_reaper_age(pid_t pid)
 	return procage;
 }
 
+/*
+ * Returns 0 on success.
+ * Complement of read_cpuacct_usage_all.
+ */
+static inline int read_cpuacct_usage_from_percpu(
+	char *cg, char *cpuset, struct cpuacct_usage *cpu_usage, int cpucount, long ticks_per_sec)
+{
+	char *usage_str = NULL;
+	char *delim = " ";
+	char *token = NULL;
+	int j = 0;
+	uint64_t usage;
+	int rv;
+
+	if (!cpu_usage) {
+		return -1;
+	}
+
+	if (!cgfs_get_value("cpuacct", cg, "cpuacct.usage_percpu", &usage_str)) {
+		rv = -1;
+		goto err;
+	}
+
+	// usage_str format, 4 cores:
+	// 120190269376599 151483821809073 145465919648605 14450503259981
+	token = strtok(usage_str, delim);
+	while (NULL != token && j < cpucount)  {
+		sscanf(token,"%lu", &usage);
+		cpu_usage[j].user = usage / 1000.0 / 1000 / 1000 * ticks_per_sec;
+		cpu_usage[j].system = 0;
+		token = strtok(NULL, delim);
```

Comment:
We can't really use `strtok` since it's not thread-safe and `libfuse` runs threaded.

Reply:
You can copy:
```c
#define lxc_iterate_parts(__iterator, __splitme, __separators)                  \
	for (char *__p = NULL, *__it = strtok_r(__splitme, __separators, &__p); \
	     (__iterator = __it);                                               \
	     __iterator = __it = strtok_r(NULL, __separators, &__p))
```
from https://github.com/lxc/lxc/blob/master/src/lxc/macro.h

Reply:
Thanks, I have fixed this problem with codes in lxc/macro.h,  at the same time add some codes to check `sscanf`'s return value.